### ISSUE TITLE
Periscope Undercover debugger

### ISSUE THREAD

**Issue by HStewartGitHome** [roelandjansen/pcmos386v501#105]
I am curios if any body knows the where about of debugger called "Periscope Undercover Debugger" or just Periscope by Brett Salter.  I use this debugger a lot with development of PC-MOS including debugging kernel with special code I believe some where in init logic.  Also with some command line for interrupts. 

Something possibly is coming up not directly related to PC-MOS but this could help.

Stewart

**Comment by stsp** [comment]
Sorry for an off-topic, but why would
you need one?
All emulators of today, have the external
debugger. Which is usually much more
reliable than the guest-side ones, and
also does not degrade the compatibility
(i.e. with most guest-side debuggers you
are limited to real-mode, or no DPMI or
alike).

**Comment by the-grue** [comment]
Back when I was actively working on this project I spent quite a bit of time looking for that bit of software or anything about it since parts/pieces are still used or referenced in the source code.  Also, in the BIN directory, PUBLIC.COM and TS.COM reference "The Periscope Company".  Not sure if related, but I haven't had any luck tracking anything down.

@HStewartGitHome was there a hardware part that went with it, do you recall?

**Comment by the-grue** [comment]
Some quick googling now that I know the actual product name and who wrote it turns up a lot of hits.  Maybe reserve that for some lunch time research...

**Comment by HStewartGitHome** [comment]
@the-grue yes it had hardware - but can be ran in software only and for pc-mos as long as you not actually in protected mode you can debug all the kernel

There is a mobile app with same name and it not related. 

Something came up with possible job that spark interest.

**Comment by the-grue** [comment]
Well, I uncovered obituaries for Mr. Salter (2016) and his wife (2017), which takes the option of asking the authors off the table.  Sad.  I did find this nugget after some searching though if you want to try it:

ftp://89.179.18.66/oldftp/TOOLS/SOFTWARE/Lang/DEBUG/PERIS531/

Not sure where it is or what it will yield but most references I have found related to Periscope Undercover Debugger list 5.31 somewhere in their text.

**Comment by HStewartGitHome** [comment]
Thanks, there is actually two modes that it can be used - it can used the hardware but also software only mode.  In theory it should also work in the PC-MOS VM.  It possible needs to patch,  I was able to rename the pgm file to zip and see contents.  

My former Boss Rod is seems to be retired or not doing much - some kind of open source medical

http://www.sunsetsystems.com/node/2

**Comment by aaronvan** [comment]
Brett Salter passed away a few years ago. Not sure if it's still maintained.

**Comment by bugrepo** [comment]
@the-grue Unfortunately, can't connect to ftp://89.179.18.66/oldftp/TOOLS/SOFTWARE/Lang/DEBUG/PERIS531/

Maybe there is another site/server to download Periscope?

**Comment by misel228** [comment]
The server is still there. So if you say you can't connect it's probably because between Jan '21 and now almost all browsers removed FTP support.

But even a regular FTP client won't help you because the "oldftp" directory has no permissions anymore so no one but "root" can access this dir.

![image](https://user-images.githubusercontent.com/727668/167265440-ee0e6682-fde6-47cb-9b40-8a524bfc42e8.png)

**Comment by bugrepo** [comment]
@misel228 Oh, I see now. I can't connect not to the server, but to `oldftp` dir...

P.S. I'm using Firefox 78 ESR, so FTP client is still there.

**Can anyone share the latest version of Periscope debugger, please?**

**Comment by the-grue** [comment]
Hey folks!  Not sure where you might get that stuff now unless @HStewartGitHome made a copy of it somewhere.  I did do some searching and was able to locate some manuals here:

https://www.os2museum.com/wp/dos/dos-library/

I'll chime in again if I find anything else related to Periscope.

**Comment by roelandjansen** [comment]
not even sure of these artifacts are still on the tapes I have here. Let's wait and see what happens -- else somewhere in my spare time I will re-read the tape(s).
